apps/webapp/app/routes/_app.orgs.$organizationSlug.projects.$projectParam.runs/route.tsx

```diff
@@ -48,6 +110,70 @@ export default function Page() {
   const isLoading = navigation.state !== "idle";
   const organization = useOrganization();
   const project = useProject();
+  const user = useUser();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const url = new URLSearchParams(location.search);
+
+  const [selectedEnvironment, setSelectedEnvironment] = useState<string>(
+    ExtendedRuntimeEnvironment.ALL
+  );
+  const [selectedStatus, setselectedStatus] = useState<string>(ExtendedJobRunStatus.ALL);
+  const [filteredList, setFilteredList] = useState(list.runs);
+
+  const handleFilterChange = (filterType: string, value: string) => {
+    url.set(filterType, value);
+
+    const filters = ["status", "environment"]; // Add more filters as needed
+    const queryString = filters
+      .map((filter) => (url.has(filter) ? `${filter}=${url.get(filter)}` : null))
+      .filter((filter) => filter !== null)
+      .join("&");
+
```

Comment:
I think you could just use `url.toString()` here. It turns a URLSearchParams object into a valid query string: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/toString